libgnucash/engine/gnc-datetime.cpp

```diff
@@ -70,22 +75,26 @@ static const TZ_Ptr utc_zone(new boost::local_time::posix_time_zone("UTC-0"));
 void _set_tzp(TimeZoneProvider& tz);
 void _reset_tzp();
 
+static boost::gregorian::date gregorian_date_from_locale_string (const std::string& str);
+
 /* To ensure things aren't overly screwed up by setting the nanosecond clock for boost::date_time. Don't do it, though, it doesn't get us anything and slows down the date/time library. */
 #ifndef BOOST_DATE_TIME_HAS_NANOSECONDS
 static constexpr auto ticks_per_second = INT64_C(1000000);
 #else
 static constexpr auto ticks_per_second = INT64_C(1000000000);
 #endif
 
-/* Vector of date formats understood by gnucash and corresponding regex
- * to parse each from an external source
+/* Vector of date formats understood by gnucash and corresponding
+ * boost string->date or regex to parse each from an external source
  * Note: while the format names are using a "-" as separator, the
  * regexes will accept any of "-/.' " and will also work for dates
  * without separators.
  */
 const std::vector<GncDateFormat> GncDate::c_formats ({
+    GncDateFormat { N_("Locale"), gregorian_date_from_locale_string },
```

Comment:
Even though we may want "Locale" as the default format, you will have to move it to the end of the list to avoid breaking saved import settings. To still get "Locale" as default in the gui, you can change the dropdown widget in the Csv Import gui to use the last GncDateFormat as default.

Reply:
Ok I didn't see that `DateFormat` is stored as an int. It would be nicer as a `string`.

Amended; however the following line doesn't seem to work:
```cpp
gtk_combo_box_set_active (GTK_COMBO_BOX(date_format_combo), GncDate::c_formats.size() - 1)
```

Reply:
I've left the default as d-m-y because I couldn't set the default combo box.

Reply:
@christopherlam If you still want to do this, assuming there are always 6 entries and the last is 'Locale' change the default in gnc-imp-settings-csv.hpp line 53 to...
            m_date_format {5}, m_currency_format {0},

Reply:
Thanks... I'd think it'd be neater to store/retrieve DateFormat as a `string` instead of `int` with a backward compatibility table so that the order becomes irrelevant. But this will be a hassle.